compiler/rustc_codegen_ssa/src/codegen_attrs.rs

```diff
@@ -80,6 +80,9 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, did: DefId) -> CodegenFnAttrs {
     let mut link_ordinal_span = None;
     let mut no_sanitize_span = None;
     for attr in attrs.iter() {
+        // Nicely bundle a call to `validate_fn_only_attr` and `fn_sig`, see above.
+        let fn_sig = || validate_fn_only_attr(attr.span).then(|| tcx.fn_sig(did));
```

Comment:
This should *probably* delay a bug, if we try looking for the fn sig but don't have it, we don't want to silently eat the attr in case the attr validation is bad.

Reply:
It does delay a bug in `validate_fn_only_attr` :D
I'll inline it.